Don't require headers when writing arrays

m1ch4ls opened the pull request:
related to #19 

I really needed just to write bunch of arrays of the same length and convert them to CSV

m1ch4ls committed:
Don't require headers when writing arrays

notslang commented on index.js:
```diff
@@ -48,7 +48,12 @@ CsvWriteStream.prototype._compile = function(headers) {
 
-  if (!isArray && !this.headers) this.headers = Object.keys(row)
+  if (!this.headers) {
+    this.headers = Object.keys(row)
```
this is kinda weird to set `this.headers` even when it's an array... maybe we should do this logic down at line 77?

m1ch4ls replied:
This seems to be required for generated functions (see `this._arrRow`) to work. To bypass the headers requirement I think we would have to rewrite the function generation.

But I might be wrong I didn't look into the code much - I just made the fix for our use case in hope it might be useful for others too...

m1ch4ls committed:
Fix generating CSV with lots of rows and cols (1000+ cols) 1M rows

Generated functions were fast, but I suspect high memory usage on our servers to be due to those functions.
